---
permalink: /
title: "üëã Hello there, I'm Tommaso"
excerpt: "Read about my research activity and interests"
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---

<p>üéì I'm a Postdoctoral Research Fellow in Information Technology @ Politecnico di Milano</p>

<p>üß¨ü¶† Doing bioinformatic is my way of trying to make the world a better place.</p>

<p>üìà I'm interested in data analysis and machine learning problems.</p>

<p>üë®‚Äçüíª I develop software instruments and methods that biology experts can use to improve our understanding of current and future viral threats.</p>

<h1>üîç Research areas in Information Technology</h1>

<h3>Anomaly detection</h3>

<table>
    <tr>
        <td><div><img src="images/about/anomaly.png" width=500px></div></td>
        <td>Application field: **Divergent viral evolution**<br><br>By integrating multidimensional anomaly detection algorithms with codon bias and dinucleotide composition, I developed a method to detect viral isolates featuring a high zoonotic risk. This method can be used to enhance genomic surveillance practices and, possibly, to anticipate pandemics.</td>
    </tr>
<!-- Appearance of table is modified by _includes/scripts.html-->

<h3>Statistical methods</h3>

<table>
    <tr>
        <td><div><img src="images/about/recomb.png" width=410px></div></td>
        <td>Application field: **Viral recombination**<br><br>Using robust statistical methods, I developed a cutting-edge method that significantly heightens the detection rate and speed of recombination events, offering a vital asset to combat future pandemics or epidemics. ([link](https://doi.org/10.1038/s41467-024-47464-5))</td>
    </tr>
<!-- Appearance of table is modified by _includes/scripts.html-->

<h3>Machine Learning</h3>

<table>
    <tr>
        <td><div><img src="images/about/gmvae.png" width=580px></div></td>
        <td>Application field: **Clustering of viruses**<br><br>Using information compression and clustering algorithms on viral sequences it is possible to highlight key epidemiological events and enable detailed comparisons of genomic features across Influenza A Virus serotypes. This approach provides a scalable framework to study viral evolution, adaptation, especially in viruses with limited genomic data available. ([link](https://doi.org/10.21203/rs.3.rs-5737660/v1))<br><br>Alternative and completely unsupervised methods, like Gaussian Mixture Variational AutoEncoders (**GMVAE**) have also been explored on few selected cases.</td>
    </tr>
<!-- Appearance of table is modified by _includes/scripts.html-->


<h3>Knowledge Integration and Management</h3>

<table>
    <tr>
        <td><div><img src="images/about/integration.png" width=430px></div></td>
        <td>Application field: **Data modeling & intgration pipelines**<br><br>Fast, reliable, incremental integration of viral sequences to develop one of the largest relational databases of viral genomes in 2020 ([ViruSurf](https://doi.org/10.1093/nar/gkaa846)). I also contributed to crafting tools facilitate the analysis of SARS-CoV- 2 virus evolution ([VirusViz](https://doi.org/10.1093/nar/gkab478), [EpiSurf](https://doi.org/10.1093/database/baab059), [CoV2K](https://doi.org/10.1038/s41597-022-01348-9)).<br><br>Application field: **AI reasoning on knowledge graphs**<br><br>I used a knowledge and reasoning representation language to bring artificial intelligence to a knowledge-base ([CoV2K](https://doi.org/10.1038/s41597-022-01348-9)), answering interesting questions about SARS-CoV-2 variants and phenotypic effcts. ([link](../files/rule_ml.pdf))</td>
    </tr>
<!-- Appearance of table is modified by _includes/scripts.html-->


<h3>Population Genomics</h3>

<table>
    <tr>
        <td><div><img src="images/about/population_gen.png" width=540px></div></td>
        <td>Application field: **Control population studies, cancer target gene identification**<br><br>I built an API to ease the selection of genomic populations and assess genomic characteristcs in the 1000 Genomes Project, and The Cancer Genome Atlas (TCGA). Combining gene annotations, this API provides summary information about any population defined with fine-grained filters on DNA variants and metadata. ([link](https://doi.org/10.1186/s12859-022-04927-0))</td>
    </tr>
<!-- Appearance of table is modified by _includes/scripts.html-->

<br>

<h1>üìö Education and Training</h1>

<ul>
    <li>PhD in Information Technology @ Politecnico di Milano, Italy</li>
    <li>Artificial Intelligence and Machine Learning course @ University of Oxford, UK</li>
    <li>Master Degree in Computer Science and Engineering @ Politecnico di Milano, Italy</li>    
</ul>

