<style>
td, th {
   border: none!important;
}
</style>


---
permalink: /
title: "üëã Hello there, I'm Tommaso"
sitemap: false
excerpt: "About me"
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---

|Anomaly Detection|Data Analysis|Knowledge Management|Population Genomics|
|-------------------|-------------------|-------------------|-------------------|
|<img src="images/about/recomb.png" width=300px>|<img src="images/about/cluster.png" width=300px>|<img src="images/about/integration.png" width=250px>|<img src="images/about/population_gen.png" width=320px>|
|**Divergent evolution** By integrating multidimensional anomaly detection algorithms with codon bias and dinucleotide composition, I developed a method to detect viral isolates featuring a high zoonotic risk, that could be used to enhance genomic surveillance and anticipate pandemics.|**Viral recombination** Using robust statistical methods, I developed a cutting-edge method that significantly heightens the detection rate and speed of recombination events, offering a vital asset in our readiness to combat future pandemics or epidemics. ([link](https://doi.org/10.1038/s41467-024-47464-5))<br><br>**Clustering of viruses** Using information compression and clustering algorithms on viral sequences it is possible to highlight key epidemiological events and enable detailed comparisons of genomic features across IAV serotypes. This approach provides a scalable framework to study viral evolution, adaptation, especially in viruses with limited genomic data available.|**Data modeling & intgration pipelines** Fast, stable and continuous integration of viral sequences to develop one of the largest integrated database of genomes in 2020 ([ViruSurf](https://doi.org/10.1093/nar/gkaa846)). I also contributed to crafting tools facilitate the analysis of SARS-CoV- 2 virus evolution ([VirusViz](https://doi.org/10.1093/nar/gkab478), [EpiSurf](https://doi.org/10.1093/database/baab059), [CoV2K](https://doi.org/10.1038/s41597-022-01348-9)).<br><br>**AI reasoning on knowledge graphs** I used a knowledge and reasoning representation language to bring artificial intelligence to a knowledge-base, answering interesting questions about SARS-CoV-2 variants and phenotypic effcts. ([link](../files/rule_ml.pdf))|**Population genomics** I built an API to ease the selection genomic populations and assess genomic characteristcs in the 1000 Genomes Project, and The Cancer Genome Atlas (TCGA). Combining gene annotations, this API provides summary information about any population defined with fine-grained filters on DNA variants and metadata. ([link](https://doi.org/10.1186/s12859-022-04927-0))|


# üñ•Ô∏è Implemented tools

- [RecombinHunt](https://github.com/tomalf2/recombinhunt-cov) - Viral recombination detection tool
- [CoV2K](http://gmql.ue/cov2k/api/) - API for a knowledge base of SARS-CoV-2 variants and effects
- [VirusViz](http://gmql.eu/virusviz) - web app for analysing mutaions of SARS-CoV-2
- [VarSum](http://gmql.eu/popstudy) - API for selecting human genomic populations 
- [ViruSurf](http://gmql.eu/virusurf_gisaid) - Search engine for viruses with specific genomic mutations or metadata requirements
- [EpiSurf](http://gmql.eu/episurf_gisaid) - Search engine for viruses dirupting epitope recognition by the immune response

# üìú Featured publications

**[Data-driven recombination detection in viral genomes](https://doi.org/10.1038/s41467-024-47464-5)** 
*Alfonsi T., Bernasconi A., Chiara M., Ceri S.* 
Nature Communications, vol. 15, n. 3313, 2024 

**[CoV2K model, a comprehensive representation of SARS-CoV-2 knowledge and data interplay](https://doi.org/10.1038/s41597-022-01348-9)** 
*Alfonsi T., Al Khalaf R., Ceri S., Bernasconi A.* 
Scientific Data, vol. 9, n. 260, 2022, pp. 1-12

**[EpiSurf: metadata-driven search server for analyzing amino acid changes within epitopes of SARS-CoV-2 and other viral species](https://doi.org/10.1093/database/baab059)** 
*Bernasconi A., Cilibrasi L., Al Khalaf R., Alfonsi T., Ceri S., Pinoli P., Canakoglu A.* 
Database, vol. 2021, 2021

**[ViruSurf: an integrated database to investigate viral sequences](https://doi.org/10.1093/nar/gkaa846)** 
*Canakoglu A., Pinoli P., Bernasconi A., Alfonsi T., Damianos P. M., Ceri S.* 
Nucleic Acids Research, vol. 29, n. D1, pp. D817-D824

**[VirusViz: comparative analysis and effective visualization of viral nucleotide and amino acid variants](https://doi.org/10.1093/nar/gkab478)** 
*Bernasconi A., Gulino A., Alfonsi T., Canakoglu A., Pinoli P., Sandionigi A., Ceri S.*
Nucleic Acids Research, vol. 49, n. 15, pp. e90

**[Genomic data integration and user-defined sample-set extraction for population variant analysis](https://doi.org/10.1186/s12859-022-04927-0)** 
*Alfonsi T., Bernasconi A., Canakoglu A., Masseroli M.* 
BMC Bioinformatics, vol. 23, n. 401, pp. 1-33

[...more](https://scholar.google.com/citations?view_op=list_works&hl=it&hl=it&user=v5lhgMMAAAAJ)

# üôã‚Äç‚ôÇÔ∏è Conferences 
- IEEE International Conference on Big Data Computing Service and Machine Learning Applications (IEEE BigDataService), 2023 & 2024; Member of Program Commitee
- Doctoral Achievements, Colloquia Doctoralia 2022, Milan, Italy
- Viral Genomics Conference 2022, Hinxton, England
- Declarative AI: International Joint Conference on Rules
and Reasoning (RuleML+RR) 2022, Berlin, Germany
- 13th International Conference on Semantic Web Applications and Tools for Health Care and Life Sciences (SWAT4HCLS), Leiden, Netherlands
- 15th Research Challenges in Information Science (RCIS), Limassol, Cyprus

# üìö Education and Training

- PhD in Information Technology @ Politecnico di Milano, Italy
- Artificial Intelligence and Machine Learning course @ University of Oxford, UK
- Master Degree in Computer Science and Engineering @ Politecnico di Milano, Italy

# üë®‚Äçüè´ Teaching

- Python programming for the MSs in Bioinformatics and Computational Genomics
- C and Matlab programming in the BSs in Mechanical Engineering